src/com/facebook/buck/python/CxxPythonExtensionDescription.java

```diff
@@ -201,12 +203,16 @@ protected Path getExtensionPath(BuildTarget target, Flavor pythonPlatform, Flavo
                 args.platformLinkerFlags,
                 cxxPlatform)));
 
-    // Embed a origin-relative library path into the binary so it can find the shared libraries.
+    // Embed a extension-relative library path into the .so so it can find the shared libraries that
+    // it depends on.
+    Path baseModule = PythonUtil.getBasePath(params.getBuildTarget(), args.baseModule);
+    String rootPath = MorePaths.relativize(Paths.get(""), baseModule).toString();
     argsBuilder.addAll(
         StringArg.from(
             Linkers.iXlinker(
                 "-rpath",
-                String.format("%s/", cxxPlatform.getLd().resolve(ruleResolver).libOrigin()))));
+                String.format("%s/%s", cxxPlatform.getLd().resolve(ruleResolver).libOrigin(),
+                rootPath))));
```

Comment:
The indentation makes it look like `rootPath` is the third argument of `Linkers.iXlinker` instead of `String.format`.

Move each argument to `String.format` on its own line.